mapie/estimators.py

```diff
@@ -190,31 +216,34 @@ def fit(self, X: ArrayLike, y: ArrayLike) -> MapieRegressor:
             The model itself.
         """
         self._check_parameters()
+        estimator = self._check_estimator(self.estimator)
+        cv = self._check_cv(self.cv)
         X, y = check_X_y(X, y, force_all_finite=False, dtype=["float64", "object"])
-        self.single_estimator_ = clone(self.estimator).fit(X, y)
+        self.n_features_in_ = X.shape[1]
+        self.estimators_ = []
```

Comment:
Pourquoi avoir déplacé `estimators_` et `k_` en dehors du `else` ? Pour pouvoir les inclure dans le `check_is_fitted` du `predict` ?

Reply:
Pour alléger le code, on comprend mieux qu'il y a une phase d'initialisation et une phase de calcul, beaucoup plus concise. De plus, cela permet d'avoir toujours les mêmes attributs, quelque soit la méthode, ce qui est moins confusant.

Reply:
Mais l'inconvénient est de déclarer des listes vides qui ne vont pas être utilisées.

Reply:
Ce n'est pas vraiment un inconvénient. Si un utilisateur a un code qui repose sur l'évaluation de `mapie.estimators_` (dans un boucle par exemple), son code fonctionnera quelque soit la méthode utilisée (et la boucle aura 0 itérations si la liste est vide). A l'inverse, si l'on n'a pas les mêmes attributs pour toutes les méthodes, son code crashera dès qu'il changera de méthode. Ce n'est pas souhaitable.